include/openmc/distribution.h

```diff
@@ -222,12 +230,28 @@ class Equiprobable : public Distribution {
   vector<double> x_; //! Possible outcomes
 };
 
-using UPtrDist = unique_ptr<Distribution>;
+//==============================================================================
+//! Mixture distribution
+//==============================================================================
+
+class Mixture : public Distribution {
+public:
+  explicit Mixture(pugi::xml_node node);
+
+  //! Sample a value from the distribution
+  //! \param seed Pseudorandom number seed pointer
+  //! \return Sampled value
+  double sample(uint64_t* seed) const;
+
+  // d and p property
+  const vector<UPtrDist>& d() const { return d_; }
+  const vector<double>& p() const { return p_; }
+private:
+  vector<UPtrDist> d_;    //!< Pointer to sub-distributions
```

Comment:
I would use `distribution_` here and `distribution()` above for the accessor to be consistent with other code (e.g., look at secondary_correlated.h)

Reply:
Ok, I will rework that. I have a new implementation with p and d more tightly couppled. This will allow to use std::lower_bound for a bisection search O(log(N)) instead O(N). Maybe an overkill, if only O(2) distributions are used...